compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -1157,6 +1157,112 @@ def IREELinalgExt_WinogradInputTransformOp : IREELinalgExt_Op<"winograd.input_tr
   }];
 }
 
+def IREELinalgExt_WinogradFilterTransformOp : IREELinalgExt_Op<"winograd.filter_transform",
+    [DeclareOpInterfaceMethods<ReifyRankedShapedTypeOpInterface>]> {
+  let summary = "Winograd Filter Transform operator";
+  let description = [{
+    Filter transform op.
+  }];
+
+  let arguments = (ins Variadic<AnyShaped>:$inputs,
+                       Variadic<AnyShaped>:$outputs,
+                       I64Attr:$output_tile_size,
+                       I64Attr:$kernel_size,
+                       DenseI64ArrayAttr:$kernel_dimensions
+  );
+
+  let builders = [
+    OpBuilder<(ins "ValueRange":$inputs, "ValueRange":$outputs,
+      CArg<"int64_t", "8">:$output_tile_size, CArg<"int64_t", "3">:$kernel_size,
+      CArg<"ArrayRef<int64_t>", "{0, 1}">:$kernel_dimensions)>
+  ];
+
+  let results = (outs Variadic<AnyRankedTensor>:$result);
+  let hasFolder = 1;
+  let assemblyFormat = [{
+    attr-dict
+    `output_tile_size` `(` $output_tile_size `)`
+    `kernel_size` `(` $kernel_size `)`
+    `kernel_dimensions` `(` $kernel_dimensions `)`
+    `ins` `(` $inputs `:` type($inputs) `)`
+    `outs` `(` $outputs `:` type($outputs) `)`
+    (`->` type($result)^)?
+  }];
+
+  let extraClassDeclaration = extraLinalgExtOpClassDeclaration # [{
+    Value input() {
+      return getDpsInputOperand(0)->get();
+    }
+    Value output() {
+      return getDpsInitOperand(0)->get();
+    }
+    ShapedType getInputOperandType() {
+      return input().getType().cast<ShapedType>();
+    }
+    ShapedType getOutputOperandType() {
+      return output().getType().cast<ShapedType>();
+    }
+    Value getOriginalOperand() {
+      return input();
+    }
+    Value getTransformedOperand() {
+      return output();
+    }
```

Comment:
We don't need them because we already have `getInput` and `getOutput`. I think anyone who works on the op should already have context of the op. So we don't need other accessor with more meaningful naming. `getInput` and `getOutput` is good enough to me.

Reply:
I added these so the filter transform will work with a unit dim folding pattern in the decomposition:
https://github.com/iree-org/iree/blob/e87ff178a9a7b210ec2c0e6726024adb0b0124e6/compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/DecomposeWinogradPass.cpp#L32-L40

I decided to add the functions here instead of having switch cases in the pass. I figured it is useful to have a function like this, since the ops are similar. Would it be better to remove them and just add a switch case in the decomposition pass?

Reply:
I see, thanks for the details!